Tools/CMake/AMReX_Config.cmake

```diff
@@ -157,6 +157,19 @@ function (configure_amrex)
       endif()
    endif()
 
+   #
+   # Setup HDF5 -- for now we do not create an imported target
+   #
+   if (ENABLE_HDF5)
+      set(HDF5_PREFER_PARALLEL TRUE)
+      find_package(HDF5 1.10.4 REQUIRED COMPONENTS CXX)
```

Comment:
I think AMReX also implements the C api, since the C++ bindings of HDF5 do not support parallel I/O (yet).
(This is not the same component naming convention as in FindMPI here.)
```cmake
find_package(HDF5 1.10.4 REQUIRED COMPONENTS C)
```